--- ./datasources/ServiceDataSource.py  2018-06-08 11:28:23.081781834 -0400
+++ ./datasources/ServiceDataSource.new 2018-06-08 11:30:20.463245858 -0400
@@ -320,6 +320,7 @@
         except Exception:
             data['events'].append({
                 'severity': ZenEventClasses.Error,
+                'eventClass': '/Status',
                 'eventClassKey': 'WindowsServiceCollectionStatus',
                 'eventKey': 'WindowsServiceCollection',
                 'summary': 'No or bad results returned for service query',
@@ -373,6 +374,7 @@
             'device': config.id,
             'summary': 'Windows Service Check: successful service collection',
             'severity': ZenEventClasses.Clear,
+            'eventClass': '/Status',
             'eventKey': 'WindowsServiceCollection',
         })
